code/modules/mob/inventory.dm

```diff
@@ -209,7 +209,11 @@
 	return FALSE
 
 //Puts the item into our active hand if possible. returns TRUE on success.
-/mob/proc/put_in_active_hand(obj/item/I, forced = FALSE, ignore_animation = TRUE)
+/mob/proc/put_in_active_hand(obj/item/I, forced = FALSE, ignore_animation = TRUE, offered=FALSE)
+	if(!offered) // check if a clicked item is in possession of someone already unless it's offered.
```

Comment:
Put in nactive hand should simply put the item in your active hand and not do any of this. Add the logic to offering instead.

Reply:
I really want to do a fancier way, but offer code isn't really compatible well with this.

Reply:
Needs to be done a better way that is more modularised and contained